src/peakprom.jl

```diff
@@ -203,7 +203,17 @@ function peakproms!(peaks::AbstractVector{Int}, x::AbstractVector{T};
                 rref = exm(view(notmval, k:rb)) # Slice corollary upper side
             end
 
-            proms[i] = abs(x[peaks[i]] - exa(coalesce(lref, rref), coalesce(rref, lref)))
+            proms[i] =  abs(x[peaks[i]] -
+                            # exa(coalesce(lref, rref), coalesce(rref, lref)))
+                            # we manually union-split this for better type-inference
+                            if ismissing(lref)
+                                rref
+                            elseif ismissing(rref)
+                                lref
+                            else
+                                exa(lref, rref)
+                            end
```

Comment:
I'm not a big fan, stylistically, of the if/else's there. Could you move this block above the line 206 and assign to a variable before using in line 206?

Reply:
Updated